web/src/DebugDemandMode.svelte

```diff
@@ -1,31 +1,76 @@
 <script lang="ts">
   import type { Feature, FeatureCollection, MultiPolygon } from "geojson";
+  import type { DataDrivenPropertyValueSpecification } from "maplibre-gl";
   import {
     FillLayer,
     GeoJSON,
     hoverStateFilter,
     LineLayer,
   } from "svelte-maplibre";
-  import { emptyGeojson } from "svelte-utils/map";
+  import { SequentialLegend } from "svelte-utils";
+  import { emptyGeojson, makeRamp } from "svelte-utils/map";
   import { SplitComponent } from "svelte-utils/top_bar_layout";
   import BackButton from "./BackButton.svelte";
-  import { layerId, Link, sum } from "./common";
+  import { layerId, Link } from "./common";
+  import { demandColorScale } from "./common/colors";
   import { backend, mode } from "./stores";
   import type { ZoneDemandProps } from "./wasm";
 
   let gj = emptyGeojson() as FeatureCollection<MultiPolygon, ZoneDemandProps>;
   try {
     gj = $backend!.getDemandModel();
-    console.log(gj);
   } catch (err) {
     window.alert("No demand model for this area");
     $mode = { mode: "pick-neighbourhood" };
   }
 
+  let showTo = false;
+
   let hovered: Feature | null = null;
   $: hoveredId = hovered == null ? null : (hovered.id as number);
   // MapLibre doesn't preserve the arrays in properties, so use the original version
   $: current = hoveredId != null ? gj.features[hoveredId] : null;
+
+  $: [limits, fillColor] = getLimitsAndColor(hoveredId, showTo);
+
+  function getLimitsAndColor(
+    hoveredId: number | null,
+    showTo: boolean,
+  ): [string[], DataDrivenPropertyValueSpecification<string>] {
+    if (hoveredId == null) {
+      let key: "sum_from" | "sum_to" = showTo ? "sum_to" : "sum_from";
+
+      let max = Math.max(...gj.features.map((f) => f.properties[key]));
+      let limits = maxIntoBuckets(max);
+
+      return [
+        limits.map((l) => l.toLocaleString()),
+        makeRamp(["get", key], limits, demandColorScale),
+      ];
+    } else {
+      // The arrays get stringified as properties and can't be used. Make an
```

Comment:
The number of zones per study area isn't too huge, so this isn't horrible.

Reply:
I'm assuming this has to do with maplibre stringifying any nested properties?

I'm not sure  I'm understanding this comment.

Is it: because we can't pass in an array of counts (due to maplibre nested properties limitations), we instead have to construct a verbose style rule that explicitly assigns the color to each neighborhood id.

Reply:
Clarified in another commit. Your understanding is correct